sdk/src/main/java/com/mapbox/maps/StyleObserver.kt

```diff
@@ -98,16 +111,22 @@ internal class StyleObserver(
         styleDataStyleLoadedListener = null
       }
       StyleDataType.SPRITE -> {
-        styleDataSpritesLoadedListener?.onStyleLoaded(
-          loadedStyle ?: throw MapboxMapException("Style is not initialized yet although SPRITES event has arrived!")
-        )
-        styleDataSpritesLoadedListener = null
+        styleDataSpritesLoadedListener?.let {
+          styleDataSpritesLoadedListener = null
+          it.onStyleLoaded(
+            loadedStyle ?: throw MapboxMapException("Style is not initialized yet although SPRITES event has arrived!")
+          )
+        }
```

Comment:
firstly, we could try make `styleDataSpritesLoadedListener` non-nullable and introduce meaningful bool flag;
secondly, this code seems to duplicate code from https://github.com/mapbox/mapbox-maps-android/pull/1378/files#r880446875 - I think it makes sense to introduce function and re-use it

Reply:
Talked offline - will extract logic with callback call to the function instead